code/modules/unit_tests/metabolizing.dm

```diff
@@ -34,3 +34,63 @@
 
 	TEST_ASSERT(!user.reagents.has_reagent(meth), "User still has meth in their system when it should've finished metabolizing")
 	TEST_ASSERT(!user.has_movespeed_modifier(/datum/movespeed_modifier/reagent/methamphetamine), "User still has movespeed modifier despite not containing any more meth")
+
+/datum/unit_test/addictions/Run()
+	var/mob/living/carbon/human/pill_user = allocate(/mob/living/carbon/human)
+	var/mob/living/carbon/human/syringe_user = allocate(/mob/living/carbon/human)
+	var/mob/living/carbon/human/pill_syringe_user = allocate(/mob/living/carbon/human)
+
+	var/obj/item/reagent_containers/pill/pill = allocate(/obj/item/reagent_containers/pill)
+	var/obj/item/reagent_containers/pill/pill_two = allocate(/obj/item/reagent_containers/pill)
+
+	var/obj/item/reagent_containers/syringe/syringe = allocate(/obj/item/reagent_containers/syringe)
+
+	var/datum/reagent/drug/methamphetamine/meth = allocate(/datum/reagent/drug/methamphetamine)
+
+
+	// Let's start with stomach metabolism
+	pill.reagents.add_reagent(meth.type, meth.addiction_threshold)
+	pill.attack(pill_user, pill_user)
+
+	// Set the metabolism efficiency to 1.0 so it transfers all reagents to the body in one go.
+	var/obj/item/organ/stomach/pill_belly = pill_user.getorganslot(ORGAN_SLOT_STOMACH)
+	pill_belly.metabolism_efficiency = 1
+
+	pill_user.Life()
```

Comment:
You'll want to pause the mobs subsystem for this test. Search `SSmobs.pause()` to see how this is done. (Note to self: Fix this boilerplate some day)

Reply:
Addressed